headers/extended_matrix.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include "extended_types.h"
+#include <unordered_map>
+
+
+/**
+* Here is implementation of a endless(within reasonable limits of course) any rank sparse matrix using unordered_map
```

Comment:
What are those limits, btw? =)

Reply:
It was just a joke, but seriously - there are always restrictions.